### PR TITLE
Add single zone

### PR DESCRIPTION
**Pull request by sgaoshang**
Add nutanix E2E test
OCP-70809 and OCP-70625
Rehearse job passed here 
https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/49940/rehearse-49940-periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-nutanix-ipi-zones-single-f28/1769752436211191808
https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/49940/rehearse-49940-periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-nutanix-ipi-zones-f14/1769903138652819456

### CODE REVIEW

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-nutanix-ipi-zones-single-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-nutanix-ipi-zones-single-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-nutanix-ipi-zones-single-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-nutanix-ipi-zones-single-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-nutanix-ipi-zones-single-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-nutanix-ipi-zones-single-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-nutanix-ipi-zones-f14

**Comment by sgaoshang** [comment]
@jianlinliu @gpei pls help to review it, thanks.

File: ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installation-nightly-4.16.yaml [hunk 1]
```diff
@@ -290,6 +290,17 @@ tests:
...
+- as: nutanix-ipi-zones-single-f28
+  cluster: build01
+  cron: 27 16 16 * *
+  steps:
+    cluster_profile: nutanix-qe-zone
+    env:
+      EXTRACT_MANIFEST_INCLUDED: "true"
+      SINGLE_ZONE: "true"
+    test:
+    - chain: openshift-e2e-test-qe
```
Comment (jianlinliu) [inline ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installation-nightly-4.16.yaml:302]: If this job is only for installation test cases, then we do not need to run e2e testing, but use `ref: cucushift-installer-check-cluster-health` to replace it. If you wan to run it with e2e testing, I think we should move it to regular ci yaml.

File: ci-operator/step-registry/ipi/conf/nutanix/zones/ipi-conf-nutanix-zones-commands.sh [hunk 2]
```diff
@@ -26,14 +24,17 @@ declare prism_element3_subnet
 # shellcheck source=/dev/null
 source "${NUTANIX_AUTH_PATH}"
 
+failureDomains="- failure-domain-1\n        - failure-domain-2\n        - failure-domain-3"
+if [ "$SINGLE_ZONE" ]; then
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/nutanix/zones/ipi-conf-nutanix-zones-commands.sh:28]:
```suggestion
if [ "$SINGLE_ZONE" ]; then
```
```suggestion
if [[ "$SINGLE_ZONE" == "true" ]]; then
```

File: ci-operator/step-registry/cucushift/installer/check/nutanix/zones/cucushift-installer-check-nutanix-zones-commands.sh [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+declare prism_element1
+declare prism_element2
+declare prism_element3
+
+# shellcheck source=/dev/null
+source "${NUTANIX_AUTH_PATH}"
+
+node_zone_list=$(oc get nodes -o=jsonpath='{.items[*].metadata.labels.topology\.kubernetes\.io\/zone}')
+
+if [ "$SINGLE_ZONE" ]; then
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/check/nutanix/zones/cucushift-installer-check-nutanix-zones-commands.sh:18]:
```suggestion
if [[ "$SINGLE_ZONE" == "true" ]]; then
```

File: ci-operator/step-registry/cucushift/installer/check/nutanix/zones/cucushift-installer-check-nutanix-zones-commands.sh [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    for zone in ${node_zone_list}; do
+        if [ "$zone" != "$prism_element1" ]; then
+            echo "Fail: fail to check single zone: $zone, expected: $prism_element1"
+            exit 1
+        fi
+    done
+else
+    # shellcheck disable=SC2076
+        if ! [[ " ${node_zone_list[*]} " =~ " $prism_element1 " ]]; then
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/check/nutanix/zones/cucushift-installer-check-nutanix-zones-commands.sh:27]:
```suggestion
      if ! [[ " ${node_zone_list[*]} " =~ " $prism_element1 " ]]; then
```

File: ci-operator/step-registry/cucushift/installer/check/nutanix/zones/cucushift-installer-check-nutanix-zones-commands.sh (hunk 3 above)
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/check/nutanix/zones/cucushift-installer-check-nutanix-zones-commands.sh:18]: Can we also print out some info about the checking is running against single zone or multi zones cluster in the if and else branch?

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.16-nutanix-ipi-zones-single-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.16-nutanix-ipi-zones-single-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.16-nutanix-ipi-zones-single-f28

**Pull request merged**